Marlin/src/sd/cardreader.cpp

```diff
@@ -222,20 +223,110 @@ void CardReader::selectByIndex(SdFile dir, const uint8_t index) {
 //
 // Get file/folder info for an item by name
 //
-void CardReader::selectByName(SdFile dir, const char * const match) {
+void CardReader::selectByName(SdFile dir, const char * const match, const bool debug/*=false*/) {
```

Comment:
runtime debug?

Looking at this code, all this debug code seems to be only useful while developing. After that, I don't think it's needed.

Reply:
Yes, that was part of a runtime debugger I had set up. 
The reason for the parameter in the routine like that was so that while I was debugging i could choose to display the debugger items in this routine for problematic operations, but not for ones that work. IE: I knew it was working fine with M20's PrintListing, but was having issues with the DiveToFile routine.  Those actual debugging lines would get stripped out of the code during compile time (except for the debug boolean being set to false) if the main debugger for that module was not defined.